### ISSUE TITLE
Pessimistic Concurrency: Backport to 3.x?

### ISSUE THREAD

**Issue by Tobias-Gr** [Particular/NServiceBus.Persistence.Sql#297]
Hi,
will the pessimistic concurrency feature from version 4 be backported to version 3.x ? We would like to use this feature with NServiceBus 6.x.

Thanks
Tobias

**Comment by SzymonPobiega** [comment]
@Tobias-Gr is there anything preventing your from upgrading to NSB 7? Since changes between 6 and 7 are really cosmetic (on the outside), the effort required should be comparable to upgrading to the next minor release.

**Comment by Tobias-Gr** [comment]
I am not aware of anything technical preventing us from upgrading to NSB 7. It's more a psychological aspect to not early adopt NSB 7 for an important next release of our product.

**Comment by SzymonPobiega** [comment]
@Tobias-Gr fair enough.

**Comment by SzymonPobiega** [comment]
@Tobias-Gr we discussed this internally and we decided it would be risky to try to retro-fit this into 3.x. Moreover, since this is a feature we would need to release a new minor release of 3.x which would mean that 4.0, in terms of features, is not a superset of 3. This would create a lot of confusion for users who would upgrade from 3.x to 4.0 and, suddenly, the performance of their saga access would drop significantly.
